server/ip-tools.ts

```diff
@@ -157,6 +161,11 @@ export const IPTools = new class {
 		const ips = range.split('-');
 		if (ips.length === 2) {
 			const [minIP, maxIP] = ips;
+
+			const minIPNumber = IPTools.ipToNumber(minIP);
+			const maxIPNumber = IPTools.ipToNumber(maxIP);
+			if (minIPNumber < 0 || minIPNumber > maxIPNumber) return false;
+
 			return this.ipRegex.test(minIP.trim()) && this.ipRegex.test(maxIP.trim());
```

Comment:
If `ipToNumber` does validation as it's intended to, the regex should be unnecessary.

```suggestion
			return minIPNumber >= 0 && minIPNumber < maxIPNumber;
```